conn_test.go

```diff
@@ -696,23 +695,19 @@ func TestErrorDuringStartupClosesConn(t *testing.T) {
 func TestBadConn(t *testing.T) {
 	var err error
 
-	bad := &atomic.Value{}
-	bad.Store(false)
-	cn := conn{bad: bad}
+	cn := conn{}
 	func() {
 		defer cn.errRecover(&err)
 		panic(io.EOF)
 	}()
 	if err != driver.ErrBadConn {
 		t.Fatalf("expected driver.ErrBadConn, got: %#v", err)
 	}
-	if !cn.getBad() {
+	if err := cn.err.get(); err == nil {
```

Comment:
we should assert what kind of error gets returned here

Reply:
fixed